### PR TITLE
feat(tracing): support minitrace + jaeger

### LINKED ISSUE #6587: feat: support minitrace-jaeger
After we migrating from tokio-tracing to minitrace, jaeger has been supported yet.

https://github.com/risingwavelabs/risingwave/blob/d725b67cf83855e09040fb88523f859770013783/src/utils/runtime/src/lib.rs#L126-L128

This should be part of https://github.com/risingwavelabs/risingwave/issues/4120 or https://github.com/risingwavelabs/risingwave/issues/951.

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR adds support for jaeger with minitrace.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- Close #6587

### CODE REVIEW

**Review by fuyufjh — DISMISSED** [review]

File: src/tracing/src/lib.rs [hunk 1]
```diff
@@ -57,27 +84,67 @@ impl RwTracingService {
...
                 let func = move || {
-                    let stream = rx.for_each_concurrent(None, |collector| async move {
+                    let rt = tokio::runtime::Builder::new_current_thread().build()?;
+                    let stream = rx.for_each_concurrent(None, |collector| async {
                         let spans = collector.collect().await;
-                        if let Some(span) = spans.first() {
-                            // print requests > 100ms
-                            if span.duration_ns >= slow_request_threshold_ms * 1_000_000 {
-                                tracing::info!("{:?}", spans);
+                        println!("{:?}", spans);
```
Comment (liurenjie1024) [inline src/tracing/src/lib.rs:118]: Use tracing::debug?

File: src/tracing/src/lib.rs [hunk 2]
```diff
@@ -13,36 +13,63 @@
...
+    pub jaeger_endpoint: Option<String>,
+}
+
 impl RwTracingService {
     /// Create a new tracing service instance. Spawn a background thread to observe slow requests.
-    #[allow(clippy::new_without_default)]
-    pub fn new() -> Self {
+    pub fn new(config: TracingConfig) -> Result<Self> {
         let (tx, rx) = unbounded();
         let slow_request_threshold_ms: u64 = env::var("RW_TRACE_SLOW_REQUEST_THRESHOLD_MS")
```
Comment (liurenjie1024) [inline src/tracing/src/lib.rs:37]: I think it's better to put these envs into `TracingConfig`

File: src/tracing/src/lib.rs [hunk 3]
```diff
@@ -13,36 +13,63 @@
...
             .ok()
             .map_or_else(|| 100, |v| v.parse().unwrap());
 
-        let join_handle = Self::start_tracing_listener(rx, slow_request_threshold_ms);
+        let jaeger_addr: Option<SocketAddr> =
+            config.jaeger_endpoint.map(|x| x.parse()).transpose()?;
 
-        Self {
+        let print_to_console = env::var("RW_TRACE_SLOW_REQUEST")
```
Comment (liurenjie1024) [inline src/tracing/src/lib.rs:44]: As above

File: src/compute/src/server.rs [hunk 4]
```diff
@@ -120,6 +121,16 @@ pub async fn compute_node_serve(
         state_store_metrics.clone(),
         object_store_metrics,
         TieredCacheMetricsBuilder::new(registry.clone()),
+        Arc::new(
+            risingwave_tracing::RwTracingService::new(TracingConfig {
+                jaeger_endpoint: if opts.enable_jaeger_tracing {
+                    Some("127.0.0.1:6831".to_string())
```
Comment (liurenjie1024) [inline src/compute/src/server.rs:127]: Make it customizable?

Comment (skyzh) [reply src/compute/src/server.rs:127]: in the future 🤣

File: src/tracing/src/lib.rs (hunk 1 above)
Comment (skyzh) [reply src/tracing/src/lib.rs:118]: I expect this to be effective in release mode

Comment (skyzh) [reply src/tracing/src/lib.rs:118]: Yeah, but we should use tracing::info for sure

**Review by fuyufjh — APPROVED** [review]
LGTM

**Pull request merged**